src/main/java/de/metas/ui/web/product/process/WEBUI_M_Product_LotNumber_Lock.java

```diff
@@ -0,0 +1,126 @@
+
+/*
+ * #%L
+ * metasfresh-webui-api
+ * %%
+ * Copyright (C) 2018 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+/**
+ * https://github.com/metasfresh/metasfresh/issues/3693
+ * This process will search for HUs containing products with LotNo attributes that fit the pairs in the selected I_M_Product_LotNumber_Lock entries.
+ * If such HUs are found, they will all be put in a DD_Order and sent to the Quarantine warehouse.
+ * 
+ * @author metas-dev <dev@metasfresh.com>
+ *
+ */
+public class WEBUI_M_Product_LotNumber_Lock extends ViewBasedProcessTemplate implements IProcessPrecondition
+{
+
+	@Override
+	protected String doIt() throws Exception
+	{
+		getView().streamByIds(getSelectedRowIds())
+				.map(row -> row.getId().toInt())
+				.distinct()
+				.forEach(id -> quarantineHUsForLotNo(id));
+		return MSG_OK;
+	}
+
+	@Override
+	protected ProcessPreconditionsResolution checkPreconditionsApplicable()
+	{
+		final DocumentIdsSelection selectedRowIds = getSelectedRowIds();
+		if (selectedRowIds.isEmpty())
+		{
+			return ProcessPreconditionsResolution.rejectBecauseNoSelection();
+		}
+
+		return ProcessPreconditionsResolution.accept();
+	}
+
+	private void quarantineHUsForLotNo(final int lotNoLockId)
+	{
+		final I_M_Product_LotNumber_Lock lotNoLock = load(lotNoLockId, I_M_Product_LotNumber_Lock.class);
+		final I_M_Attribute lotNumberAttr = Services.get(ILotNumberDateAttributeDAO.class).getLotNumberAttribute(getCtx());
+
+		if (lotNumberAttr == null)
+		{
+			throw new AdempiereException("Not lotNo attribute found.");
+		}
+
+		final int productId = lotNoLock.getM_Product_ID();
+		final int lotNoAttributeID = lotNumberAttr.getM_Attribute_ID();
+		final String lotNoValue = lotNoLock.getLot();
+
+		final List<I_M_HU> husForAttributeStringValue = retrieveHUsForAttributeStringValue(productId, lotNoAttributeID, lotNoValue);
+
+		Services.get(IHUDDOrderBL.class).createQuarantineDDOrderForHUs(husForAttributeStringValue);
+
+	}
+
+	private List<I_M_HU> retrieveHUsForAttributeStringValue(final int productId, final int attributeId, final String value)
+	{
+		return Services.get(IHandlingUnitsDAO.class).createHUQueryBuilder()
+				.addOnlyWithProductId(productId)
+				.list()
+				.stream()
+				.filter(hu -> hasHUAttributeAndValue(hu, attributeId, value))
```

Comment:
why not using de.metas.handlingunits.IHUQueryBuilder.addOnlyWithAttribute(I_M_Attribute, Object) ?
it would be much more efficient that retrieving the whole database and checking the HUs one by one which has the quarantine flag set.

Reply:
also, now coming to my mind: does HUQueryBuilder by default filter by HU status?
at any rate, i suggest to make it explicit and filter by the states which we care for
* not planned, shipped or destroyed
* what about picked/issued maybe for those we need to skip them and show a user message?
  * @metas-mk wdyt?

Reply:
exactly, only active/ picked handling units (only those which are still in our warehouse). the other ones we have to get out of retraceability (other issue).

Reply:
done